datafusion/functions-aggregate-common/src/aggregate/groups_accumulator.rs

```diff
@@ -348,6 +352,46 @@ impl GroupsAccumulator for GroupsAccumulatorAdapter {
     fn size(&self) -> usize {
         self.allocation_bytes
     }
+
+    fn convert_to_state(
+        &self,
+        values: &[ArrayRef],
+        opt_filter: Option<&BooleanArray>,
+    ) -> Result<Vec<ArrayRef>> {
+        let num_rows = values[0].len();
+
+        // Each row has its respective group
+        let mut results = vec![];
+        for row_idx in 0..num_rows {
+            // Create the empty accumulator for converting
+            let mut converted_accumulator = (self.factory)()?;
```

Comment:
as a follow on PR I wonder if we could potentially to improve performance by adding a `clear()` or `reset()` type function to each accumulator to avoid having to create a new accumulator for each group.

Reply:
Yes, I want to reuse the `converted_accumulator` at the beginning, but it is not ensure that the state will be reset after calling `state`. 
It is clever to add such function to do the reset work explicitly.